flow/scripts/synth.tcl

```diff
@@ -50,6 +53,16 @@ if {[info exist ::env(PRESERVE_CELLS)]} {
   }
 }
 
+
+
+if {[info exist ::env(BLOCKS)]} {
+  hierarchy -check -top $::env(DESIGN_NAME)
+  foreach block $::env(BLOCKS) {
+    blackbox $block
+    puts "blackboxing $block"
+  }
+}
+
```

Comment:
PRESERVE_CELLS variable is shared between top-level and macro-level, therefore adding BLOCKS to it might result in issues in macro-level and therefore they had to be separated, I'll try reorganizing this part though.